boot/BootFromDevice.c

```diff
@@ -94,6 +99,9 @@ CONFIGENTRY *AddNestedConfigEntry(CONFIGENTRY *config, CONFIGENTRY *nested, char
 	memset(config, 0x00, sizeof(CONFIGENTRY));
 	strncpy(config->title, title, sizeof(config->title));
 	config->nestedConfigEntry = nested;
+	config->bootType = nested->bootType;
+	config->drive = nested->drive;
+	config->partition = nested->partition;
```

Comment:
Not sure this addition is needed, since nested entries purpose is only to show the boot menu.